src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,53 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_SIZE = 10;
+    private Object[] keys = new Object[STORAGE_SIZE];
+    private Object[] values = new Object[STORAGE_SIZE];
+    private int size;
+    private int index;
+
+    public StorageImpl() {
+
+    }
+
+    public StorageImpl(Object[] keys, Object[] values, int size) {
+        this.keys = keys;
+        this.values = values;
+        this.size = size;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (hasAKey(key)) {
+            values[index] = value;
+            return;
+        }
+        keys[size] = key;
+        values[size] = value;
+        size++;
     }
 
     @Override
     public V get(K key) {
+        if (hasAKey(key)) {
+            return (V) values[index];
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        return size;
+    }
+
+    private boolean hasAKey(K key) {
+        for (int i = 0; i < size; i++) {
+            index = i;
+            if ((key == keys[i]) || key != null && key.equals(keys[i])) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
Using class variable to hold some state should be used only when really necessary as such decision requires additional monitoring of it's state and may cause unintentional behaviour.
Here we may easily return index instead of keeping it in class variable:
```suggestion
    private int getKeyIndex(K key) {
        for (int i = 0; i < size; i++) {
            if ((key == keys[i]) || key != null && key.equals(keys[i])) {
                return i;
            }
        }
        return -1;
```